rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,503 @@
+
+```yaml
+publicWebhook: false 
+...
+Additional Configuration Items.
+...
+```
+
+---
+- ## Installation YAML
+
+The user can pass a yaml file along with the `--addon-config` flag to override the default settings we provide.
+```sh
+linkerd install --addon-config config.yaml | kubectl apply -f -
+```
+
+The yaml file will be unmarshalled during installation and will overwrite the default values in `values.yaml`
+
+The template of the `yaml` file can be defined as follows 
+
+```yaml
+# Flag to enable/disable sending alerts to Webhook.site
+publicWebhook: true [ default = false ]
+
+# This section contains configuration items specific to Alertmanager addon component
+alertmanager:
+
+  # Name of the Alertmanager component, re-used across helm chats 
+  name: alertmanager [ default = alertmanager ]
+
+  # Flag to determine if Alertmanager should be enabled/disabled.
+  enabled: true [ default = false ]
+
+  # Image name of a stable release of Alertmanager, that works best with the current version of Prometheus
+  image: prom/alertmanager:v0.19.0 [ default = prom/alertmanager:v0.19.0 ]
+
+  # Path to the file containing the configMap for Alertmanager Deployment
+  config: path/to/config/file.yaml [ default = "" ]
+
+# This section contains configuration items specific to Prometheus component
+prometheus:
+
+  # Path to file containing Prometheus Rules
+  rules: path/to/rules/file.yaml [ default = "" ]
+
+  # Path to the swagger spec file that will be used to generate a service profile and provide per route metrics
+  swaggerSpecPath: path/to/swagger/spec [ default = "" ]
+
+  # Path to the protobuf spec file that will be used to generate a service profile and provide per route metrics
+  protobufSpecPath: path/to/protobuf/spec [ default = "" ]
+```
+
+
+---
+- ## Prometheus Helm Chart
+The Alertmanager installation will come out of the box with some Prometheus rules, and having a `config.yaml` will allow the user to override the default rules. 
+
+The rule file that is mentioned in `config.yaml` will need to be read, unmarshalled and then embedded in the Prometheus ConfigMap. 
+
+We need to inform Prometheus about the existence of a Alertmanager service and this is done by adding an alerting mechanism in the Prometheus ConfigMap in `charts/linkerd2/templates/prometheus.yaml` file 
+
+```yaml
+alerting:
+  alertmanagers:
+  - scheme: http
+    static_configs:
+    - targets: ['{{.Values.alertmanager.name}}:9093']
+```
+
+As part of `linkerd install` we can also check the syntax of the Prometheus Rules that are passed in the `config.yaml` file using a [**utility**](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/#syntax-checking-rules) that Prometheus provides. This will allow us to gracefully exit in case there are syntax errors in the Prometheus rules.  
```

Comment:
If we are supporting the path link model, we don't really have to do this, as its a separate file and folks can definitely check if the rules are right by running the tool.

if we are going with the full config in `addon-config` model, we can do this and it shouldn't be hard but I don't really think it should be a priority.

Reply:
Do you suggest I remove it from the RFC?